target_benchmark/tasks/AbsTask.py

```diff
@@ -412,15 +414,19 @@ def _update_retrieval_metrics(
         """
         num_queries = len(new_retrieval_results)
         for idx in range(num_queries):
-            db_id = query_batch[DATABASE_ID_COL_NAME][idx]
-            table_id = query_batch[TABLE_ID_COL_NAME][idx]
+            gold_db_id: str = query_batch[DATABASE_ID_COL_NAME][idx]
+            gold_table_id: Union[str, List[str]] = query_batch[TABLE_ID_COL_NAME][idx]
             retrieval_result = new_retrieval_results[idx]
-            if table_id == "N/A" or not table_id:
-                if str(db_id) in [result[0] for result in retrieval_result.retrieval_results]:
-                    self.true_positive += 1
-            else:
-                if (str(db_id), str(table_id)) in retrieval_result.retrieval_results:
-                    self.true_positive += 1
+            if not isinstance(gold_table_id, list):
+                # Treat all datasets (even single-table settings) as a list
+                gold_table_id = [gold_table_id]
+            normalized_gold_tables = set(
+                (gold_db_id, t) for t in gold_table_id
+            )  # E.g. {('soccer_3', 'club'), ('soccer_3', 'players')}
+            self.num_overlap += len(normalized_gold_tables.intersection(set(retrieval_result.retrieval_results)))
```

Comment:
I made a minor update to this line, previously it had the `=` operator and I changed it to `+=`. 
when it was `=` it seemed all recalls were 0, likely because the assignment will keep `self.num_overlap` at a very small number.
@parkervg could you confirm that this is the intended behavior?

Reply:
@jixy2012 Yes exactly, your change is correct! Great catch!